bundles/org.openhab.binding.webexteams/src/main/java/org/openhab/binding/webexteams/internal/WebexTeamsHandler.java

```diff
@@ -0,0 +1,315 @@
+                return;
+            }
+        } else {
+            this.status = ThingStatus.OFFLINE;
+            updateStatus(this.status, ThingStatusDetail.CONFIGURATION_ERROR, "@text/confErrorTokenOrId");
+            return;
+        }
+
+        // background initialization:
+        scheduler.execute(() -> {
+            try {
+                this.client = new WebexTeamsApi(this, oAuthFactory, httpClient);
+                logger.debug("Trying to fetch account details");
+                Person p = this.client.getPerson();
+                logger.debug("Success: {}", p.getDisplayName());
+
+                this.status = ThingStatus.ONLINE;
+                updateStatus(this.status);
+
+                refresh();
+
+            } catch (NotAuthenticatedException e) {
+                logger.error("Failed to initialize client", e);
+                this.status = ThingStatus.OFFLINE;
+                updateStatus(this.status, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            }
+        });
+
+        // TODO: make interval configurable.
+        refreshService = scheduler.scheduleWithFixedDelay(this::refresh, 0, 300, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing...");
+        final ScheduledFuture<?> job = refreshService;
+        if (job != null) {
+            job.cancel(true);
+            refreshService = null;
+        }
+        this.client.dispose();
+    }
+
+    // mainly used to refresh the auth token when using OAuth
+    private void refresh() {
+        if (status == ThingStatus.ONLINE) {
+            Person person;
+            try {
+                person = client.getPerson();
+                updateState(CHANNEL_BOTNAME, StringType.valueOf(person.getDisplayName()));
+                updateProperty("personType", person.getType());
+                updateProperty("name", person.getDisplayName());
+
+                // Only when the identity is a person:
+                if (person.getType().equalsIgnoreCase("person")) {
+                    String status = person.getStatus();
+                    updateState(CHANNEL_STATUS, StringType.valueOf(status));
+                    DateFormat df = new SimpleDateFormat(ISO8601_FORMAT);
+                    String lastActivity = df.format(person.getLastActivity());
+                    updateState(CHANNEL_LASTACTIVITY, new DateTimeType(lastActivity));
+                }
+            } catch (WebexTeamsApiException e) {
+                logger.warn("Failed to update display name: {}", e.getMessage());
+            }
+
+        }
+    }
+
+    /**
+     * Sends a message to the default room.
+     *
+     * @param msg markdown text string to be sent
+     *
+     * @return <code>true</code>, if sending the message has been successful and
+     *         <code>false</code> in all other cases.
+     */
+    public boolean sendMessage(String msg) {
+
```

Comment:
Other example here but not the last one.

Reply:
Fixed.